airbyte-integrations/connectors/destination-iceberg/src/main/java/io/airbyte/integrations/destination/iceberg/config/catalog/NessieCatalogConfig.java

```diff
@@ -0,0 +1,98 @@
+  private final String ref;
+  private final String authenticationType;
+  private final String nessieApiVersion;
+  private final String token;
+
+  public NessieCatalogConfig(@NotNull JsonNode catalogConfig) {
+    Preconditions.checkArgument(null != catalogConfig.get(NESSIE_CATALOG_URI_CONFIG_KEY), "%s is required", NESSIE_CATALOG_URI_CONFIG_KEY);
+    JsonNode tokenNode = catalogConfig.get(NESSIE_CATALOG_TOKEN_CONFIG_KEY);
+    JsonNode clientApiNode = catalogConfig.get(NESSIE_CLIENT_API_VERSION_KEY);
+    JsonNode refNode = catalogConfig.get(NESSIE_CATALOG_REFERENCE_KEY);
+    JsonNode authTypeNode = catalogConfig.get(NESSIE_CATALOG_AUTHENTICATION_TYPE_KEY);
+    this.token = null != tokenNode ? tokenNode.asText() : null;
+    this.nessieApiVersion = null != clientApiNode ? clientApiNode.asText() : null;
+    this.uri = catalogConfig.get(NESSIE_CATALOG_URI_CONFIG_KEY).asText();
+    this.ref = null != refNode ? refNode.asText() : null;
+    this.authenticationType = null != authTypeNode ? authTypeNode.asText() : null;;
+  }
+
+  @Override
+  public Map<String, String> sparkConfigMap() {
+    Map<String, String> configMap = new HashMap<>();
+    configMap.put("spark.network.timeout", "300000");
+    configMap.put("spark.sql.defaultCatalog", CATALOG_NAME);
+    configMap.put("spark.sql.extensions", "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME, "org.apache.iceberg.spark.SparkCatalog");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".catalog-impl", "org.apache.iceberg.nessie.NessieCatalog");
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".uri", this.uri);
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".ref", this.ref);
+    configMap.put("spark.sql.catalog." + CATALOG_NAME + ".authentication.type", this.authenticationType);
+    configMap.put("spark.driver.extraJavaOptions", "-Dpackaging.type=jar -Djava.io.tmpdir=/tmp");
+
+    //adds all the S3 related configurations.
+    Preconditions.checkArgument(this.storageConfig instanceof S3Config, "Nessie Catalog only supports S3 file system.");
+    configMap.putAll(this.storageConfig.sparkConfigMap(CATALOG_NAME));
+    return configMap;
+  }
+
+  @Override
+  public Catalog genCatalog() {
+    NessieCatalog catalog = new NessieCatalog();
+    Map<String, String> properties = new HashMap<>(this.storageConfig.catalogInitializeProperties());
+    properties.put(CatalogProperties.URI, this.uri);
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, this.storageConfig.getWarehouseUri());
+    if (isNotBlank(this.ref)) {
+      properties.put("ref", this.ref);
+    }
+    if (isNotBlank(this.authenticationType)) {
+      properties.put(NESSIE_CATALOG_AUTHENTICATION_TYPE_KEY, this.authenticationType);
+    }
+    if (isNotBlank(this.nessieApiVersion)) {
+      properties.put("client-api-version", this.nessieApiVersion);
```

Comment:
does the `client-api-version` really matter that much from a user's perspective? I would assume that it would be ok to just use the latest version by default, without having a config for this

Reply:
While API v1 is deprecated in Nessie, the Iceberg `NessieCatalog` still [defaults to v1](https://github.com/apache/iceberg/blob/39239a19555f55e0facc0a585086791a606a4b32/nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java#L104). I think it makes sense to keep this config to allow users to opt in for v2, until `NessieCatalog` is updated to default to v2.

We should probably default to Nessie API v2 in Airbyte.

Reply:
Actually, `NessieCatalog` in Iceberg 1.3.0 does not support choosing the API version. It will always use v1. Perhaps we have to defer adding this config property to a later version, when newer Iceberg artifacts are integrated with Airbyte.

Reply:
@nk1506 : WDYT about moving `client-api-version` out of this PR and re-adding it later, when Iceberg is upgrades in Airbyte?

Reply:
I believe this needs to be address, because exposing this config setting to end users can be confusing when the API version is not actually configurable with the current set of dependencies (Iceberg's `NessieCatalog`).